compiler/rustc_const_eval/src/const_eval/machine.rs

```diff
@@ -236,6 +236,16 @@ impl<'mir, 'tcx> interpret::Machine<'mir, 'tcx> for CompileTimeInterpreter<'mir,
 
     const PANIC_ON_ALLOC_FAIL: bool = false; // will be raised as a proper error
 
+    #[inline(always)]
+    fn enforce_alignment(ecx: &InterpCx<'mir, 'tcx, Self>) -> bool {
+        ecx.tcx.sess.opts.unstable_opts.extra_const_ub_checks
+    }
+
+    #[inline(always)]
+    fn enforce_validity(ecx: &InterpCx<'mir, 'tcx, Self>) -> bool {
+        ecx.tcx.sess.opts.unstable_opts.extra_const_ub_checks
+    }
```

Comment:
a bit confused, did we not have an impl for these functions before? Afaict the trait definition doesn't actually define these functions?

https://github.com/rust-lang/rust/blob/8d1fa7105b2ad3154f63d8f0641fd7da87e20e8f/compiler/rustc_const_eval/src/interpret/machine.rs#L133

Reply:
For reference, this is https://github.com/rust-lang/rust/issues/71614.